python/mooseutils/gitutils.py

```diff
@@ -72,18 +72,28 @@ def git_submodule_info(working_dir=os.getcwd(), *args):
     """
     Return the status of each of the git submodule(s).
     """
+    command = ['git', 'submodule', 'status', *args]
+    try:
+        result = check_output(command, cwd=working_dir, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        # If the status command failed, it's possible that some submodules are broken, e.g., cannot
+        # be recursed because it is empty, so run a non-initializing update and try again.
+        subprocess.call(['git', 'submodule', 'update', '--recursive'], cwd=working_dir)
```

Comment:
I don't think we want to do this automatically, because it can cause problems with compiling. Some application are made up of other apps, that are made up of others. However, we use the MOOSE from the top-level. If we init all submodules we can end up downloading multiple copies of MOOSE/libmesh/petsc, etc.

What problem are you trying to correct?

Reply:
@aeslaughter It's a very obscure problem and I explained it in item 2 of issue #17951. It's possible for a sub-submodule to be initialized, but in a totally empty `(null)` state, especially `moose/large_media`, at which point it is impossible to recurse and Git throws an error. This is why I added the check for `(null)` tags in `git_init_submodule()`. 

I agree this might be pretty unsafe, I suppose I'll just revert this code back.